aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+"""EddsaJcs2022 cryptosuite."""
+
+from hashlib import sha256
+import canonicaljson
+
+from ....wallet.base import BaseWallet
+from ....wallet.keys.manager import MultikeyManager
+from ....utils.multiformats import multibase
+from ....core.profile import ProfileSession
+from ....resolver.did_resolver import DIDResolver
+from ..errors import PROBLEM_DETAILS
+
+
+class CryptosuiteError(Exception):
+    """Generic Cryptosuite Error."""
+
+
+class EddsaJcs2022:
+    """EddsaJcs2022 cryptosuite.
+
+    https://www.w3.org/TR/vc-di-eddsa/#eddsa-jcs-2022.
+    """
+
+    def __init__(self, *, session: ProfileSession):
+        """Create new EddsaJcs2022 Cryptosuite instance.
+
+        Args:
+            session: ProfileSession to use during crypto operations.
+
+        """
+        super().__init__()
+        self.session = session
+        self.wallet = session.inject(BaseWallet)
+        self.key_manager = MultikeyManager(session)
+
+    async def create_proof(self, unsecured_data_document: dict, options: dict):
```

Comment:
I think I'd like to see a model representing options rather than a plain dictionary.

Reply:
I can do this yes, good suggestion.

Reply:
DIOptions model added